Sources/Parsing/Parsers/Many.swift

```diff
@@ -74,7 +80,11 @@ where
     var rest = input
     var result = self.initialResult
     var count = 0
-    while count < self.maximum, let output = self.upstream.parse(&input) {
+    while
+      count < self.maximum,
+      let output = self.upstream.parse(&input),
+      memcmp(&input, &rest, MemoryLayout<Upstream.Input>.size) != 0
+    {
       count += 1
       rest = input
       self.updateAccumulatingResult(&result, output)
```

Comment:
We could potentially `memcmp` on the `result` instead...but that might create copies...

Reply:
Would it be a problem if `updateAccumulatingResult` doesn't actually modify `result` but still consumes input? eg. you might ignore a subset of the parser's output?

Reply:
That too!